backend/dao/winesDAO.js

```diff
@@ -0,0 +1,87 @@
+const ObjectID = require("mongodb").ObjectID;
+
+let wines;
+
+module.exports = class WinesDAO {
+  static async injectDB(connection) {
+    if (wines) {
+      return;
+    }
+
+    try {
+      wines = await connection.db(process.env.USER_NS).collection("wines");
+    } catch (e) {
+      console.log(`Unable to establish connection in winesDAO: ${e}`);
+    }
+  }
+
+  static async getWines({ filters = null, page = 0, winesPerPage = 5 } = {}) {
+    let query;
+    //!console.log("filters ", filters);
+    //!console.log(filters.price);
+
+    if (filters) {
+      if (filters.price == "low") {
+        //todo:figure out the price ranges!
+        query = { price: { $gte: 0, $lt: 30.0 } };
+      } else if (filters.price == "med") {
+        //todo:figure out the price ranges!
+        query = { price: { $gte: 30.0, $lt: 100.0 } };
+      } else if (filters.price == "high") {
+        //todo:figure out the price ranges!
+        query = { price: { $gte: 100 } };
+      } else if ("food" in filters) {
+        //*This filter is set up in the mongodb atlas interface, creating an index.
+        query = { $text: { $search: filters["food"] } };
+      }
+      let cursor;
+      try {
+        cursor = await wines.find(query);
+      } catch (e) {
+        console.error(`Unable to issue find query: ${e}`);
+        return { wineList: [], totalWines: 0 };
+      }
+      const displayCursor = cursor
+        .limit(winesPerPage)
+        .skip(winesPerPage * page);
+      try {
+        const winesList = await displayCursor.toArray();
+        //!console.log("wine list: ", winesList);
+        const totalWines = await wines.countDocuments(query);
+        console.log("total wines:", totalWines);
+        return { winesList, totalWines };
+      } catch (e) {
+        console.error(`Unable to convert cursor to array: ${e}`);
+        return { wineList: [], totalWines: 0 };
+      }
+    }
+  }
+
+  static async addWine(
+    name,
+    country,
+    food,
+    price,
+    type,
+    vegan,
+    image_url,
+    wineProfile
+  ) {

```

Comment:
Usually when we have 3 or more parameters in a functions, we should start considering to make them an object parameter, just like the `getWines`, because it gets confusing to know exactly which parameter is which position.
It depends by case, but for 8 parameters which have the same importance I'd say it's better to have as object.

Reply:
Oh, thank you, you answered my questions I had while writing it.. I will modify that

Reply:
Good to hear!
Yeah, when it's 3-4, it's still debatable, but 5 or more it's very likely better to make it an object.

Reply:
I hope I modified it correctly- I sent the req object as the parameter